packages/sdk/src/stores/google-datastore/store.ts

```diff
@@ -1,15 +1,18 @@
+import { exists } from "../../utils";
 import type { SortedStore } from "..";
-import type { Datastore, Key } from "@google-cloud/datastore";
+import { Datastore, Key } from "@google-cloud/datastore";
```

Comment:
Before we get too deep into integrating Datastore (since we haven't done much work on it so far), should we use firestore rather than datastore? By my understanding, firestore is basically just the next generation of datastore and Google is trying to migrate users.

See https://cloud.google.com/datastore (they have a blurb about firestore being the next generation of google datastore).

Edit: the library you're using _may_ work with both?

Reply:
I believe this is right. firestore is their newer noSQL data base (datastore was released in 2008 while firestore was 2017). Firestore looks to have more modern syntax, better documentation and more language support. This article is helpful in comparing the two: https://cloud.google.com/datastore/docs/firestore-or-datastore. you can actually run fire store in datastore mode

however, when I try to access firestore from our GCP account I get this:
![image](https://user-images.githubusercontent.com/12886084/130020624-49f04c94-c6c3-4061-9858-89204d495355.png)

indicating you need to use one _or_ the other in gcp 🤔 . this might be a per-project restriction, which is not an issue as we can make a separate project for the API (@evaldofelipe might have already set this up!)

Reply:
@mrice32 regarding using the same library if you use firestore in "native" mode then you will need to use this sdk: https://github.com/googleapis/nodejs-firestore if you use firestore in "datastore" mode then you are correct we can keep using https://github.com/googleapis/nodejs-datastore. 

this is mentioned at the bottom of the readme for the `nodejs-firestore` package:

_Note: This Cloud Firestore Server SDK does not support Firestore databases created in Datastore mode. To access these databases, use the Datastore SDK._

Reply:
wow this is really good to know. would be happy to try firestore instead

Reply:
Let me adjust the project automation to support that first.

Reply:
im going to leave this PR as a google datastore implementation, and work on a firestore implmementation as a separate feature, most likely going to use thier updated client https://googleapis.dev/nodejs/firestore/latest/
https://app.clubhouse.io/uma-project/story/2119/add-google-firestore-store-in-sdk

Reply:
Either way works. It seems they recommend firestore in datastore mode for "server" applications, so we might want to stick with that and keep the current client implementation.

@daywiss however, if you find that you like the firestore sdk more, then we can swap to native mode!

Reply:
after our discussion offline, i think running firestore in datastore mode is right. that means this client should just work out of the box. will run some tests once we have it available in gcp